File: src/main/java/com/university/entities/Department.java
package com.university.entities;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

@Entity
@Table(name = "department")
public class Department {

    @Id
    @Column(name = "Id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @Column(name = "Name", nullable = false, length = 30)
    private String name;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "department_position",
            joinColumns = @JoinColumn(name = "Department_Id"),
            inverseJoinColumns = @JoinColumn(name = "Position_Id")
    )
    private List<Position> positions = new ArrayList<>();

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Position> getPositions() {
        return positions;
    }

    public void setPositions(List<Position> positions) {
        this.positions = positions;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Department that = (Department) o;
        return id == that.id &&
                Objects.equals(name, that.name) &&
                Objects.equals(positions, that.positions);
    }

    @Override
    public int hashCode() {

        return Objects.hash(id, name, positions);
    }

    @Override
    public String toString() {
        return "Department{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", positions=" + positions +
                '}';
    }
}

File: src/main/kotlin/com/university/entities/Department.kt
package com.university.entities

import javax.persistence.*

@Entity
@Table(name = "department")
class Department {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id : Int? = null

    @Column(name = "name", nullable = false, length = 30)
    var name: String = ""

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "department_position", joinColumns = [JoinColumn(name = "department_Id")], inverseJoinColumns = [JoinColumn(name = "position_Id")])
    var positions: List<Position> = mutableListOf()
}
